File: HashMap-Dictionary/LongestSubstring.py
class TwoSum:
    def longest_consecutive(self, nums):
        numSet = set(nums)
        longestStreak = 0

        for num in numSet:
            if (num-1) not in numSet:
                current_num = num
                current_streak = 1

                while (current_num+1) in numSet:
                    current_num += 1
                    current_streak += 1

                longestStreak = max(longestStreak, current_streak)
        return longestStreak
    
if __name__ == "__main__":
    nums = [100, 4, 200, 1, 3, 2]
    result = TwoSum()
    output = result.longest_consecutive(nums)
    print(output)

File: HashMap-Dictionary/LongestSubstring.java
import java.util.*;
class LongestSubstring{
   static int longestconsecutive(int[] nums){
        Set<Integer> numSet = new HashSet<>();
        for(int num : nums){
            numSet.add(num);
        }

        int longeststreak = 0;
        for(int num : numSet){
            if(!numSet.contains(num-1)){
                int current = num;
                int currentstreak = 1;

                while(numSet.contains(current + 1)) // keep checking for other numbers
                {
                    current = current + 1;
                    currentstreak = currentstreak + 1;
                }
                longeststreak = Math.max(longeststreak, currentstreak);
            }
        }
        return longeststreak;
    }

    public static void main(String[] args) {
        int[] nums = {100, 4, 200, 1, 3, 2};
        int result = longestconsecutive(nums);
        System.out.println(result);
    }
}